### PR TITLE
Implementing Transaction model

### PR DESCRIPTION
**Pull request by erdimaden**
### What changed? Why?
- Implementing new Transaction Model
- Adding `parseUnsignedPayload` util function to parse the unsigned payloads

### CODE REVIEW

File: src/coinbase/transaction.ts
Comment (alex-stone) [inline src/coinbase/transaction.ts:21]: Is it sufficient that this is typed via typescript? Or is this just guarding against a nil value?

Main reason we added that to the ruby client is for better error messaging when passing in the wrong object, because there is no typing.

Comment (erdimaden) [reply src/coinbase/transaction.ts:21]: The guard in the constructor is mainly to catch null values and give better error messages, similar to what we did in the Ruby client. While TypeScript handles type checking, having this check is still useful for making the code more robust and the errors clearer. If someone uses the SDK in plain JavaScript, this kind of check is even more important since JavaScript doesn't have built-in type checking.

File: src/coinbase/tests/transaction_test.ts
Comment (John-peterson-coinbase) [inline src/coinbase/tests/transaction_test.ts:63]: Lets move these out of the individual `it` blocks. Its not necessary to duplicate them if the `model` passed into the constructor is the same.

File: src/coinbase/transaction.ts
Comment (John-peterson-coinbase) [inline src/coinbase/transaction.ts:19]: we use private constructor in most places. Is there a reason this diverges?

File: src/coinbase/transaction.ts
Comment (John-peterson-coinbase) [inline src/coinbase/transaction.ts:54]:
```suggestion
  /**
   * Returns the Status of the Transaction.
```

File: src/coinbase/transaction.ts
Comment (John-peterson-coinbase) [inline src/coinbase/transaction.ts:63]:
```suggestion
   * Returns the From Address ID for the Transaction.
```

File: src/coinbase/transaction.ts
Comment (John-peterson-coinbase) [inline src/coinbase/transaction.ts:65]:
```suggestion
   * @returns The From Address ID
```

File: src/coinbase/transaction.ts
Comment (John-peterson-coinbase) [inline src/coinbase/transaction.ts:83]:
```suggestion
   * Returns the link to the Transaction on the blockchain explorer.
   *
   * @returns The link to the Transaction on the blockchain explorer
```

File: src/coinbase/transaction.ts
Comment (John-peterson-coinbase) [inline src/coinbase/transaction.ts:74]:
```suggestion
   * @returns Whether the Transaction is in a terminal State
```

File: src/coinbase/transaction.ts
Comment (John-peterson-coinbase) [inline src/coinbase/transaction.ts:94]:
```suggestion
   * @returns The ethers.js Transaction object
```

**Review by John-peterson-coinbase — COMMENTED** [review]
Lets aim to be consistent with current implementation wrt documentation (Reference: https://github.com/coinbase/coinbase-sdk-nodejs/blob/master/src/coinbase/transfer.ts)

We can choose to do these as follow ups if necessary to unblock staking team, but I would like them to be fast followed before the release if possible!

cc: @alex-stone

File: src/coinbase/tests/transaction_test.ts
Comment (alex-stone) [inline src/coinbase/tests/transaction_test.ts:50]: #initialize is a ruby construct, should this be just like `describe('constructor', ....`?

File: src/coinbase/tests/transaction_test.ts
Comment (John-peterson-coinbase) [inline src/coinbase/tests/transaction_test.ts:5]:
```suggestion
describe("Transaction", () => {
```

It seems from other projects using just class name is most idiomatic.
Lets clean these up in a follow up in other files to use `ClassName` to describe top level block.
`#InstanceMethodName` to describe instance methods and `.staticMethodName` to describe static methods.

File: src/coinbase/tests/transaction_test.ts
Comment (John-peterson-coinbase) [reply src/coinbase/tests/transaction_test.ts]: Lets keep the same structure but remain idiomatic to the language we are implementing in. `initialize` is ruby specific in this case. `Transaction` object is constructed with `constructor` in Typescript.

File: src/coinbase/tests/transaction_test.ts
Comment (erdimaden) [reply src/coinbase/tests/transaction_test.ts]: Update this we will have the most of test case refactoring/changes with [PSDK-245]

**Review by John-peterson-coinbase — APPROVED** [review]

File: CHANGELOG.md
Comment (John-peterson-coinbase) [inline CHANGELOG.md:10]: Lets format the CHANGELOG.md before merge

Comment (John-peterson-coinbase) [reply CHANGELOG.md:10]: We also should add the transaction class addition to change log

Comment (erdimaden) [reply CHANGELOG.md:10]: ah, just merged but I'm going to format in my PR

Comment (John-peterson-coinbase) [reply CHANGELOG.md:10]: no worries. Lets just change before the release.

**Pull request merged**
